addons/im_livechat/static/src/core/web/transcript_sender_patch.js

```diff
@@ -0,0 +1,33 @@
+import { TranscriptSender } from "@im_livechat/core/common/transcript_sender";
+import { ActionPanel } from "@mail/discuss/core/common/action_panel";
+import { onWillStart } from "@odoo/owl";
+import { _t } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+import { patch } from "@web/core/utils/patch";
+
+patch(TranscriptSender.prototype, {
+    setup() {
+        super.setup();
+        this.notificationService = useService("notification");
+        this.orm = useService("orm");
+        onWillStart(async () => {
+            if (this.props.thread.correspondent.persona.type == "guest") {
+                return;
+            }
+            const [result] = await this.orm.read(
+                "res.partner",
+                [this.props.thread.correspondent.persona.id],
+                ["email"]
+            );
```

Comment:
I really don't like querying the server every time we open the dialog, especially since it's only useful the first time. We should just send the email along with the correspondent's persona store data. Also, the autocompletion could already be in the base component, as it makes sense even in embed context when the visitor is a known partner.

Reply:
Yes the condition for email in store of partner is not smart and we have a task to fix it.
Fields should be requested contextually, not broadly requested and then poped unconditionally in store method.